service/history/timerQueueProcessorBase.go

```diff
@@ -578,6 +585,35 @@ func (t *timerQueueProcessorBase) processDeleteHistoryEvent(task *persistence.Ti
 	return t.deleteWorkflowVisibility(task)
 }
 
+func (t *timerQueueProcessorBase) processArchiveHistoryEvent(task *persistence.TimerTaskInfo) (retError error) {
+
+	context, release, err := t.cache.getOrCreateWorkflowExecution(t.getDomainIDAndWorkflowExecution(task))
+	if err != nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	msBuilder, err := loadMutableStateForTimerTask(context, task, t.metricsClient, t.logger)
+	if err != nil {
+		return err
+	} else if msBuilder == nil || msBuilder.IsWorkflowExecutionRunning() {
+		// this can happen if workflow is reset.
```

Comment:
I don't believe the comment is correct.  Reset always create a new run, we never reuse the run when doing reset.